gdc_ng_models/models/audit.py

```diff
@@ -0,0 +1,28 @@
+"""Common audit metadata for our data models."""
+
+from sqlalchemy import schema
+from sqlalchemy.sql import sqltypes
+
+
+class AuditColumnsMixin:
+    """A mixin to add audit columns to a data model.
+
+    Attributes:
+        created_date: Metadata describing when a row was created.
+        updated_date: Metadata describing when a row was last updated.
+    """
+
+    def __init__(self):
+        pass
+
+    created_date = schema.Column(
+        sqltypes.DateTime(timezone=True),
+        nullable=False,
+        server_default='now()'
```

Comment:
These either need to be quoted as `text()` or replaced with references to `func.now`. Otherwise, the DDL emitted by SQLAlchemy will invoke the Postgres`now()` function when it creates the table, and that specific point in time will be fixed as the default:
```
automated_test=# \d study_rule
                                              Table "public.study_rule"
    Column    |           Type           |                                 Modifiers                                  
--------------+--------------------------+----------------------------------------------------------------------------
 created_date | timestamp with time zone | not null default '2020-04-16 18:46:01.438277+00'::timestamp with time zone
 updated_date | timestamp with time zone | not null default '2020-04-16 18:46:01.438277+00'::timestamp with time zone
 id           | integer                  | not null default nextval('study_rule_id_seq'::regclass)
 name         | text                     | not null
```
This is a [weird property of SQLAlchemy](https://docs.sqlalchemy.org/en/13/core/defaults.html#server-defaults).

Reply:
good catch @singergr. its similar to setting a date column to something like `datetime.now()` the executed value will just stick around for the lifespan of the app.